tests/test_model_signature.py

```diff
@@ -4,6 +4,12 @@
 from typing_extensions import Annotated
 
 from pydantic import BaseModel, Extra, Field, create_model
+from pydantic.version import compiled
+
+# Cython 3.x uses stringified annotations by default. When a signature is generated
+# for Pydantic models, we use the `BaseModel.__init__` signature which has a type
+# of `Any` for the data. This annotation is reused and will be stringified only when compiled:
+Any_ann = "'Any'" if compiled else 'Any'
```

Comment:
This is a bit over my head, but the comment does clear things up a bit. Thanks for the note here.

Reply:
Yeah it's gross, but couldn't find a better way.